### PR TITLE
feat(router): add key_value to locker metrics

### PR DESCRIPTION
**Pull request by ShankarSinghC**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

add key_vaule to locker metrics to identify the request made to different locker (Rust or Basilisk HS).

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Testing is not required as this is just updating the key_value of the existing locker metrics.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by sai-harsha-vardhan — DISMISSED** [review]

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]
Looks like there is no counter to know successful addition, can we add that metric with appropriate attributes?

File: crates/router/src/core/payment_methods/cards.rs [hunk 1]
```diff
@@ -248,22 +256,42 @@ pub async fn add_card_to_locker(
             )
             .await
             .map_err(|error| {
-                metrics::CARD_LOCKER_FAILURES.add(&metrics::CONTEXT, 1, &[]);
+                metrics::CARD_LOCKER_FAILURES.add(
+                    &metrics::CONTEXT,
+                    1,
+                    &[router_env::opentelemetry::KeyValue::new("locker", "rust")],
```
Comment (NishantJoshi00) [inline crates/router/src/core/payment_methods/cards.rs:262]: should we add operation here too?

File: crates/router/src/core/payment_methods/cards.rs [hunk 2]
```diff
@@ -290,12 +318,16 @@ pub async fn get_card_from_locker(
             .change_context(errors::ApiErrorResponse::InternalServerError)
             .attach_printable("Failed while getting card from basilisk_hs")
             .map_err(|error| {
-                metrics::CARD_LOCKER_FAILURES.add(&metrics::CONTEXT, 1, &[]);
+                metrics::CARD_LOCKER_FAILURES.add(
+                    &metrics::CONTEXT,
+                    1,
+                    &[router_env::opentelemetry::KeyValue::new("locker", "rust")],
+                );
```
Comment (NishantJoshi00) [inline crates/router/src/core/payment_methods/cards.rs:325]: and here?

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]

File: crates/router/src/core/payment_methods/cards.rs (hunk 1 above)
Comment (ShankarSinghC) [reply crates/router/src/core/payment_methods/cards.rs:262]: Yeah, I will add it.

**Review by likhinbopanna — APPROVED** [review]

**Pull request merged**